src/main/java/seedu/address/logic/parser/tag/AddCommandParser.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.parser.tag;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONTACT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SALE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.tag.AddCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+public class AddCommandParser implements Parser<AddCommand> {
+
+    @Override
+    public AddCommand parse(String userInput) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(userInput, PREFIX_SALE, PREFIX_CONTACT, PREFIX_TAG);
+
+        System.out.println(userInput);
+        if (!arePrefixesPresent(argMultimap, PREFIX_TAG) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    seedu.address.logic.commands.tag.AddCommand.MESSAGE_USAGE));
+        }
+
+        String tagName = argMultimap.getValue(PREFIX_TAG).get();
+
+        if (tagName.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    seedu.address.logic.commands.tag.AddCommand.MESSAGE_USAGE));
+        }
+
+        Tag tag = new Tag(tagName);
+
+        if (!argMultimap.getValue(PREFIX_SALE).isEmpty()) {
+            return new AddCommand(tag, false);
+        } else if (!argMultimap.getValue(PREFIX_CONTACT).isEmpty()) {
+            return new AddCommand(tag, true);
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        }
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
P.S. I think I extracted this out into a method in `ParserUtil`. Maybe can import it from there instead of duplicating?